gatsby-node.js

```diff
@@ -685,8 +685,16 @@ exports.sourceNodes = async ({
 }) => {
   const { createNode } = actions
 
-  const data = [{ title: "Docs", path: "/" }]
-  if (glob.sync(`${process.env.LOCAL_PROJECT_DIRECTORY}/blog/**`).length > 0) {
+  const blogFiles = glob.sync(`${process.env.LOCAL_PROJECT_DIRECTORY}/blog/**`)
+  const docFiles = glob.sync(`${process.env.LOCAL_PROJECT_DIRECTORY}/**/*.md`, {
+    ignore: blogFiles,
+  })
+
+  const data = []
+  if (docFiles.length > 0) {
+    data.push({ title: "Docs", path: "/" })
+  }
+  if (blogFiles.length > 0) {
     data.push({ title: "Blog", path: "/blog" })
   }
```

Comment:
There is a pattern here. We should consider abstracting this way.
I might be wrong, but I believe in the near future we may want to support other "tracks" for the site, not just blogs and docs.

I will leave @macdonst to weigh on this.

Reply:
With that in mind, the setting of that `ignore: blogFiles` will become more complex over time. Something to think about.

Reply:
I think that is a great idea but a bit out of scope for this PR where we just need to roll out blog only support ASAP. You are 💯 correct we'll need to refactor this as we get into supporting other tracks.